src/main/scala/rise/core/primitives.scala

```diff
@@ -208,6 +208,11 @@ object primitives {
       )
   }
 
+  @primitive case class ToMem()(override val t: Type = TypePlaceholder)
```

Comment:
should this be part of core? if so, then maybe `OclToMem` could be merged with this. For example if backends could specify their memory options using a type. `C::Memory: Heap | Stack`, `OCL::Memory: ..`

Reply:
I thought about this. But since we are not making use of differentiating between Heap and Stack yet, I think we would just carry information around that we do not need at this point.

I do agree that the differentiation feels odd, but we do the same copy of primitives for all the other primitives that we have in core and which need an additional address space parameter in OpenCL (i.e.., OpenCLReduceSeq, OpenCLSlideSeq, OpenCLIterate).

I therefore suggest that we keep prefixed OpenCL primitives for now even though it could be nicer and see whether address spaces become relevant for C at some point.

Reply:
Could you open an issue about this? I think having both `reduceSeq`, `toMem` and `oclReduceSeq`, `oclToMem` when using the OpenCL backend for example is not very nice. In my opinion the primitives should either be merged, or more cleanly separated (`reduceSeq` and `toMem` would be different primitives in the C namespace/backend and in the OpenCL namespace/backend).

Reply:
I created [Issue #26](https://github.com/rise-lang/shine/issues/26) in the `shine` repo.